### PR TITLE
Lesson student list

### LINKED ISSUE #8: api/v1/teachers/:teacher_id/students endpoint
The following requests must be functional from this endpoint:

- [ ] GET all students by specific teacher ID

- [ ] POST creates new students for a specific teacher

### PR DESCRIPTION
**Pull request by Lithnotep**
## Description
Completes User Story #
Resolves #

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## Notes
Adds Single Lessonstudent endpoint
adds All LessonStudents for a give Lesson endpoint

## RSpec Results
100% passing

### CODE REVIEW

File: api/tests/test_views.py [hunk 1]
```diff
@@ -194,5 +194,39 @@ def test_post_student_lesson(self):
...
+
+    response = self.client.get('/api/v1/lessonstudents/%s/' % self.lesson.id, content_type='application/json')
+    
+    self.assertEqual(response.status_code, 200)
+    self.assertEqual(response.data[0]['score'], 5)
+    self.assertEqual(response.data[0]['mood'], 'great!')
+    self.assertEqual(response.data[0]['student'], self.student2.id)
+    self.assertEqual(response.data[1]['score'], 7)
+    self.assertEqual(response.data[1]['mood'], 'Bad!')
+    self.assertEqual(response.data[1]['student'], self.student3.id)
```
Comment (Lithnotep) [inline api/tests/test_views.py]: Forgot to pull out comment at 209, but can keep it if we want to change how the endpoint is used.

File: api/views.py [hunk 2]
```diff
@@ -68,8 +67,13 @@ def post(self, request, pk):
...
 
 
-class LessonStudent(APIView):
+class LessonStudentDetail(APIView):
   parser_classes = [JSONParser]
+
+  def get(self, request, pk, pk2):
+    get_id = LessonStudent.objects.filter(lesson_id=pk).filter(student_id=pk2).values('id')[0]['id']
+    lessonstudent = LessonStudent.objects.get(pk=get_id)
+    return Response(LessonStudentSerializer(lessonstudent).data)
```
Comment (dborski) [inline api/views.py:76]: Nice job finding this bug

File: api/tests/test_views.py (hunk 1 above)
Comment (dborski) [reply api/tests/test_views.py]: Testing looks good

File: api/tests/test_views.py [hunk 3]
```diff
@@ -194,5 +194,39 @@ def test_post_student_lesson(self):
...
+
+    response = self.client.get('/api/v1/lessonstudents/%s/' % self.lesson.id, content_type='application/json')
+    self.assertEqual(response.status_code, 200)
+    # self.assertEqual(response.data[0]['score'], 5)
+    self.assertIsInstance(response.data[0]['score'], int)
+    # self.assertEqual(response.data[0]['mood'], 'great!')
+    # self.assertEqual(response.data[0]['student'], self.student2.id)
+    self.assertIsInstance(response.data[0]['score'], int)
+    # self.assertEqual(response.data[1]['mood'], 'Bad!')
+    # self.assertEqual(response.data[1]['student'], self.student3.id)
```
Comment (melatran) [inline api/tests/test_views.py:232]: Tests look nice

File: api/views.py (hunk 2 above)
Comment (melatran) [reply api/views.py:76]: Thanks for fixing this.

**Review by melatran — APPROVED** [review]

**Review by dborski — APPROVED** [review]

**Pull request merged**